images/kernel/README.md

```diff
@@ -0,0 +1,41 @@
+# Kernel Images
+
+These kernel OCI images contain the kernel binary (at `/boot/vmlinux`) and supporting modules (in `/lib/modules`)
+for guest VMs ran by Ignite.
+
+## Building the Kernel Images
+
+```console
+$ make
+```
+
+## Versions
+
+All LTS versions starting from 4.14 and above are supported by the Ignite team.
+This means in practice:
+
+- 4.14.x
+- 4.19.x
+- 5.4.x
+
+The exact patch versions may be found in the [Makefile](Makefile).
+The available versions exist in the [stable kernel git tree](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/refs/).
+
+## Upgrading to a new kernel version
+
+It should be as easy as:
```

Comment:
It's best not to suggest something is easy/obvious in our docs.
This is a pretty magical command -- it's not simple to determine what it is doing and what the necessary order of operations is for somebody to maintain these configs and directories.

```suggestion
The kernel Makefile has an `upgrade` command that will generate patched kernel configs for each specific version in `KERNEL_VERSIONS`.

The linux kernel source code is checked out in a build container for each target version, and a resulting "defconfig" based on each Firecracker recommended base-config in the `upstream/` directory is then output to the `generated/` directory.

Once `make upgrade-%` has generated a firecracker base-config for all `KERNEL_VERSIONS`, the resulting generated configs are copied to the `versioned/`dir and are patched with ignite specific `./config-patches`. These configs under `versioned/` are used for the matching kernel builds.

Run:
```

Reply:
Good pointer, thanks!